javascript/README.md

```diff
@@ -306,6 +306,30 @@ var mydiv = $(".js-find-me");
 var mydiv = $(".find-me");
 ```
 
+#### Promises
+Prefer `async/await` over `Promise.prototype.then/catch/finally`
```

Comment:
Would you mind adding a concise description of why this is beneficial?

Without some description, there's no justification as to why it's preferred, and I'm the type of person who likes to know that so I'm not following rules blindly.

(I know many previous patterns in this file are missing a "why", but I personally don't think that is ideal)

e.g.

> Prefer `async/await` over `Promise.prototype.then/catch/finally` because it's easier to write, read, debug, and test, and it's an effective way to write asynchronous code that appears synchronous.

Reply:
good point! I added exactly what you suggested haha it summarizes the main points that came up in that Slack thread